x-pack/plugins/reporting/server/export_types/common/pdf/index.ts

```diff
@@ -124,37 +126,48 @@ export class PdfMaker {
     this._title = title;
   }
 
-  generate() {
+  async generate() {
     const docTemplate = _.assign(
       getTemplate(this._layout, this._logo, this._title, tableBorderWidth, assetPath),
       {
         content: this._content,
       }
     );
-    this._pdfDoc = this._printer.createPdfKitDocument(docTemplate, getDocOptions(tableBorderWidth));
+    this._pdfDoc = await pdfmake.createPdf(docTemplate, getDocOptions(tableBorderWidth));
```

Comment:
I assume this is the reason we want to use v0.3.0 of `pdfmake`, right? It looks like the changelog in the master branch is saying that 0.3 changes the methods to return a promise instead of taking a callback.

Maybe this means that in 0.2.4 there is a method that can take a callback, which we could use as the async method?

Reply:
Well spotted! I looked at the source code and I didn't see how the `createPdfKitDocument` could be made async. Perhaps there is another to use it...